batchflow/models/torch/base_mixins.py

```diff
@@ -221,6 +222,197 @@ class OptimalBatchSizeMixin:
 
     For stable measurements, we make `n` iterations of `train`/`predict`, until the memory consumption stabilizes.
     """
+
+    def is_cuda_out_of_memory(self, exception):
+        """ Check if exception is CUDA OOM """
+        return (
+            isinstance(exception, RuntimeError)
+            and len(exception.args) == 1
+            and "CUDA" in exception.args[0]
+            and "out of memory" in exception.args[0]
+        )
+
+    # not sure if it's necessary
+    def is_cudnn_snafu(self, exception):
+        return (
+            isinstance(exception, RuntimeError)
+            and len(exception.args) == 1
+            and "cuDNN error: CUDNN_STATUS_NOT_SUPPORTED." in exception.args[0]
+        )
+
+    def is_out_of_cpu_memory(self, exception):
+        return (
+            isinstance(exception, RuntimeError)
+            and len(exception.args) == 1
+            and "DefaultCPUAllocator: can't allocate memory" in exception.args[0]
+        )
+
+    def is_oom_error(self, exception):
```

Comment:
The entire error parsing part can be simplified to just a few lines in one method